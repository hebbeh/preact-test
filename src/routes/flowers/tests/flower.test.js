import { h } from 'preact';
import Flower from '../flower';
import { shallow } from 'preact-render-spy';
import render from 'preact-render-to-json';

const mockData = [
	{
		id: 'flower5',
		name: 'Koffing',
		description: 'Description',
		color: '#EE6055',
		info: 'Info'
	}
];

describe('Initial Test of Flower', () => {

	let wrapper, wrapperEmpty;

	beforeEach(() => {
		wrapperEmpty = shallow(
			<Flower />
		);

		wrapper = shallow(
			<Flower flower={mockData[0]} />
		);
	});

	xit('Does not render without flower-data', () => {
		expect(wrapperEmpty.length).toEqual(0);
	});

	it('Renders correctly', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('Has correct name and text', () => {
		expect(wrapper.find('h1').text()).toBe('Koffing');
		expect(wrapper.find('.flowerDesc').text()).toBe('Description');
	});

	it('Has correct initial UI', () => {
		const tree = render(<Flower flower={mockData[0]} />);
		expect(tree).toMatchSnapshot();
	});

	it('Show info when open: true', () => {
		expect((wrapper.find('.plus')).length).toEqual(1);

		wrapper.setState({ open: true });
		expect((wrapper.find('.info')).length).toEqual(1);
	});

	it('Show selectedFlower when selected: true', () => {
		expect((wrapper.find('.plus')).length).toEqual(1);

		wrapper.setState({ selected: true });
		expect((wrapper.find('.selectedFlower')).length).toEqual(1);
	});

	it('Calling handleFold will set open: true', () => {
		expect((wrapper.find('.plus')).length).toEqual(1);

		(wrapper.find('.plus')).simulate('click');
		expect((wrapper.find('.info')).length).toEqual(1);
	});

	it('Calling handleClick will set selected: true', () => {
		expect((wrapper.find('.plus')).length).toEqual(1);

		(wrapper.find('.container')).simulate('click');
		expect((wrapper.find('.selectedFlower')).length).toEqual(1);
	});
});