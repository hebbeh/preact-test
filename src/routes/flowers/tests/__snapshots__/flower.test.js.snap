// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial Test of Flower Has correct initial UI 1`] = `
<div
  class="home"
>
  <div
    class="container"
    onClick={[Function]}
  >
    <div
      class="flowerText"
    >
      <h1>
        Koffing
      </h1>
      <p
        class="flowerDesc"
      >
        Description
      </p>
      <span
        class="plus"
        onClick={[Function]}
      >
        <svg
          attr={
            Object {
              "viewBox": "0 0 512 512",
            }
          }
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
          />
        </svg>
      </span>
    </div>
    <div
      class="flower"
      style={
        Object {
          "border": "36px solid #EE6055",
        }
      }
    />
  </div>
</div>
`;
